### ISSUE TITLE
Fix HTML endpoints

### ISSUE THREAD

**Issue by glennrfisher** [watson-developer-cloud/ios-sdk#264]
The HTML endpoints return unexpected data. We need to investigate this issue to determine whether it is a problem with the service, our requests, or the JSON parsing.

Note that we have modified the HTML-related tests for AlchemyVision to fail. Unfortunately, the service returns empty strings for the model properties, which are sufficient to map to the models. Using `XCTFail()` ensures that we flag these tests and remember to return to them.

**Comment by vherrin** [comment]
from the alchemy internal chat - The entities call works without the “url” but the Microformat call throws the invalid url error. This is a bug thanks for catching it … we will add it to the backlog to fix.

**Comment by vherrin** [comment]
FYI - This may not make it in before we release this milestone

**Comment by Hsaylor** [comment]
I guess an update on this, generally:

For html and text calls the Swagger docs list the wrong information. They list the parameter as `html_File` and `text_File`, respectively, but in reality the parameters should be `html` and `text`. This is why we kept getting `content-empty` errors.

Glenn and I also figured out that there was an issue with how we were doing our encoding. Certain characters were not being encoded correctly leading to some odd issues, notably whenever an html file contained the string `url=`. If it was not encoded correctly, the service actually tried to pick it up as a query parameter. We've updated to use a custom allowed character set which seems to solve the problem for the most part.
